packages/ckeditor5-autosave/src/autosave.js

```diff
@@ -213,22 +224,8 @@ export default class Autosave extends Plugin {
 	 * @private
 	 */
 	_save() {
-		const version = this.editor.model.document.version;
-
-		// Change may not produce an operation, so the document's version
-		// can be the same after that change.
-		if (
-			version < this._lastDocumentVersion ||
```

Comment:
It seems to me that this condition can never be met. At most `version` can be the same as `_lastDocumentVersion` but it can never be lower, as `_lastDocumentVersion` is set to it, and document version never goes down.

Reply:
Yup, as I thought it was at some point `<=` (which also should simply be `==`) and it was changed to `<` because of markers that affect data but does not bump version: 5577aa2#diff-f3cd7394728f0a6168a108ff56dc6ee631d0a71eb34f1a762264f9339de28311L161

But we don't need to check versions anyway -- `_save()` is called (debounced) only in `change:date` events, so we can be sure that there was actual change.

Reply:
BTW. TBH I don't know what was the use case but if a marker affects data it should also use operations :thinking: .

While use-operation, not-data may make sense, the other combination does not. If marker affects data it has to be send over to other clients.

Reply:
However, we need to check version lower, in the promise resolution:

`if ( this.editor.model.document.version > this._lastDocumentVersion ) {`

And there might be a problem there with the markers that don't bump version but:

1.  This change (removing `if`) is not related to the code below.
2.  As I explained, I am not sure what kind of marker affects data but does not use operations :thinking: .